datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -188,7 +188,7 @@ impl<S: SimplifyInfo> ExprSimplifier<S> {
     /// assert_eq!(expr, b_lt_2);
     /// ```
     pub fn simplify(&self, expr: Expr) -> Result<Expr> {
-        Ok(self.simplify_with_cycle_count(expr)?.0)
+        Ok(self.simplify_with_cycle_count_transformed(expr)?.0.data)
```

Comment:
Only changed the `simplify_with_cycle_count` API, keep the `simplify` API to avoid influencing one more API.

Reply:
Maybe after we merge this one in, it would be worth considering changing the simplify API as well -- but I agree that would be good as a follow on pR